homeassistant/components/bluetooth/passive_update_coordinator.py

```diff
@@ -259,6 +259,15 @@ def _async_handle_bluetooth_event(
             )
             return
 
+        if not isinstance(new_data, PassiveBluetoothDataUpdate):
+            self.last_update_success = False  # type: ignore[unreachable]
```

Comment:
I think we should raise an exception or just blow up below. This would be a developer error and not something that can be fixed during runtime.

Reply:
Good idea. Will adjust after the PR I create on `bluetooth` to address the review comments in https://github.com/home-assistant/core/pull/75531

Reply:
Actually this changes how I test the other code so I'll do it now

Reply:
https://github.com/home-assistant/core/pull/75536